provider/core/src/datagen/mod.rs

```diff
@@ -90,9 +90,18 @@ pub trait DataExporter: Sync {
 /// A [`DynamicDataProvider`] that can be used for exporting data.
 ///
 /// Use [`make_exportable_provider`](crate::make_exportable_provider) to implement this.
-pub trait ExportableProvider: IterableDynamicDataProvider<ExportMarker> + Sync {}
+pub trait ExportableProvider: IterableDynamicDataProvider<ExportMarker> + Sync {
```

Comment:
Thought: Another approach here would be to make the wrapper implement `AnyProvider` instead, or _maybe_ give a blanket impl of `AnyProvider` for `ExportableProvider`

Reply:
We need concrete types to downcast and clone.

Reply:
Yeah, you're right, since ExportableProvider could be implemented on top of a buffer.